java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -896,6 +897,10 @@ public long getVersion(String tag) {
     }
   }
 
+  public SqlQuery sql(String sql, String tableName) {
```

Comment:
In rust, we make the `tableName` an optional arg. If users do not specify, the default name is `dataset`.